### PR TITLE
Populi api2 integration

### PR DESCRIPTION
**Pull request by AlecKap**
__x__ Wrote Tests __x__ Implemented __x__ Reviewed

## Necessary checkmarks:

- [x] All Tests are Passing in all environments

- [x] The code will run locally

## Type of change

- [x] New feature
- [x] Bug Fix
- [x] Refactor

## Description of Change:

    description: Convert Present app to integrate a new version of Populi API including updating all API calls, unit tests, and integration tests. 

- Removed api wrapper gem as it was used to interact with the Populi legacy api.
- Updated all methods in Populi service to make direct calls to Populi API V2.
- Created new api calls to retrieve additional appropriate data.
- Refactored any methods that were dependent on Populi api service calls to account for API V2 data structures.
- Refactored/updated tests to account for api V2 method changes.
- Replaced XML response fixtures with JSON response fixtures to match data structures of API V2.
- Implemented VCR so as to monitor and verify stability and any changes/updates to Populi API V2.

## Requested feedback:

    I'd like feedback on...

## Check the correct boxes

- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes

- [ ] No Tests have been changed
- [x] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:

- [x] My code has no unused/commented out code
- [x] My code has no binding.pry's
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link:

### CODE REVIEW

**Review by domingusj — COMMENTED** [review]
Not a huge deal, but always a good idea to remove `WIP` commits before a PR. If the commit makes sense to merge it with an adjacant one, that's a good way to squash it, or sometimes the commit still makes sense and you can just remove `WIP` from the commit message. Probably not worth fixing right now and it's a pretty nit-picky thing that some people don't care about - but just something you may run into w/ PRs

File: app/controllers/user/attendances_controller.rb [hunk 1]
```diff
@@ -58,7 +58,7 @@ def destroy
       logger.info("Attendance #{attendance.id} deleted by user #{current_user.email}. Deleted Attendance Details: #{attendance_details.inspect}")
     end
     redirect_to turing_module_path(module_id)
-    flash[:success] = "Attendance successfully deleted."
+    flash[:sucess] = "Attendance successfully deleted."
```
Comment (domingusj) [inline app/controllers/user/attendances_controller.rb]: typo?

File: app/models/zoom_meeting.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ def self.from_meeting_details(meeting_url)
     raise invalid_error if meeting_details[:code] == 3001
     raise no_meeting_error if meeting_details[:code] == 2300
     raise personal_meeting_error if meeting_details[:start_time].nil?
-
+
```
Comment (domingusj) [inline app/models/zoom_meeting.rb]: extra whitespace

File: spec/features/attendances/zoom_aliases_correction_spec.rb [hunk 3]
```diff
@@ -194,10 +193,10 @@
     student = create(:setup_student) # Need to create one setup student so the module will be considered "set up"
     other_module = student.turing_module
     other_zoom_id = "12345"
-
-    stub_request(:post, ENV['POPULI_API_URL']).
-      with(body: {"task"=>"getCourseInstanceMeetings", "instanceID"=>other_module.populi_course_id}).
-      to_return(status: 200, body: File.read('spec/fixtures/populi/course_meetings.xml'))
+    stub_course_meetings_nil
+    # stub_request(:post, ENV['POPULI_API_URL']).
```
Comment (domingusj) [inline spec/features/attendances/zoom_aliases_correction_spec.rb]: can remove commented lines if not needed

File: spec/features/modules/setup/account_match_spec.rb [hunk 4]
```diff
@@ -1,37 +1,31 @@
...
       within '#best-match' do
         click_button 'Yes'
       end
-
+      
       @anthony_b = @mod.students.find_by(name: "Anthony Blackwell Tallent")
       @j = @mod.students.find_by(name: "J Seymour")
       @leo = @mod.students.find_by(name: "Leo Banos Garcia")
-
+
```
Comment (domingusj) [inline spec/features/modules/setup/account_match_spec.rb]: extra whitespaces

**Review by domingusj — APPROVED** [review]

File: app/controllers/user/attendances_controller.rb (hunk 1 above)
Comment (AlejandroLopez1992) [reply app/controllers/user/attendances_controller.rb]: This is fixed. Thank you!

File: app/models/zoom_meeting.rb (hunk 2 above)
Comment (AlejandroLopez1992) [reply app/models/zoom_meeting.rb]: This is fixed. Thank you!

File: spec/features/attendances/zoom_aliases_correction_spec.rb (hunk 3 above)
Comment (AlejandroLopez1992) [reply spec/features/attendances/zoom_aliases_correction_spec.rb]: This is fixed. Thank you!

File: spec/features/modules/setup/account_match_spec.rb (hunk 4 above)
Comment (AlejandroLopez1992) [reply spec/features/modules/setup/account_match_spec.rb]: This is fixed. Thank you!

**Pull request merged**